# Context
You have been hired for a new job as a Data Analyst.

* The company is called "TechStream Solutions", and the product is a Software as a Service (SaaS) platform named "Streamline Pro". This platform provides comprehensive project management and collaboration tools for businesses of all sizes.

* TechStream Solutions has been operating for several years and has gathered significant data on their costs and revenues. They are now looking to analyze their unit economics to understand the profitability of Streamline Pro on a per-customer basis.

* The datasets are in the shared folder on Google Drive:
https://drive.google.com/drive/folders/1qhOW9Y2orRXuzbX-kXEmuJ7TMQiRs2Uv?usp=drive_link

* By performing these calculations, TechStream Solutions aims to:

  * Identify the profitability of acquiring and retaining customers.
  * Assess the efficiency of their marketing and sales strategies.
  * Make informed decisions on scaling their operations and optimizing their resource allocation.
  * This information will guide TechStream Solutions in refining their business strategies, ensuring sustainable growth, and maximizing profitability.

1. CAC: Example Calculation
2. Average Revenue Per User (ARPU)
3. COGS
4. Gross Margin
5. LTV
6. LTV / CAC
7. Conlusion

In [ ]:
import pandas as pd


## 1. Customer Acquisition Cost (CAC)

$$ {CAC} = \frac {TotalSalesandMarketingExpenses}{NumberofNewCustomersAcquired}$$

In [ ]:
monthly_expenses = pd.read_excel(
    'https://docs.google.com/spreadsheets/d/10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw/export?format=xlsx',
    sheet_name='Sheet1')
monthly_expenses.head(5)

#      month           category                  item  amount
0  1 2023-01-01       Server Costs           AWS Hosting    8000
1  2 2023-01-01       Server Costs  Google Cloud Storage    4000
2  3 2023-01-01  Software Licenses        Atlassian Jira    1200
3  4 2023-01-01  Software Licenses                 Slack     800
4  5 2023-01-01  Software Licenses            Salesforce    1500

In [ ]:
monthly_expenses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   #         27 non-null     int64         
 1   month     27 non-null     datetime64[ns]
 2   category  27 non-null     object        
 3   item      27 non-null     object        
 4   amount    27 non-null     int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 1.2+ KB


In [ ]:
last_month_expenses = monthly_expenses[monthly_expenses['month'] == monthly_expenses['month'].max()]
last_month_expenses.head(5)

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700

In [ ]:
crm_expenses = last_month_expenses[last_month_expenses['item'] == 'Salesforce']['amount'].values[0]
crm_expenses

np.int64(1700)

In [ ]:
#totalexpenses
totalexpenses = crm_expenses
totalexpenses

np.int64(1700)

In [ ]:
payroll = pd.read_excel('https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/export?format=xlsx',
                        sheet_name='Sheet1')
payroll.head(5)

month department  employee_name           position  paid
0 2023-01-01      Sales       John Doe      Sales Manager  1500
1 2023-01-01      Sales     Jane Smith    Sales Associate   600
2 2023-01-01      Sales      Jim Brown    Sales Associate   700
3 2023-01-01      Sales   Laura Miller    Sales Associate   800
4 2023-01-01  Marketing  Alice Johnson  Marketing Manager  1650

In [ ]:
last_month_payroll = payroll[payroll['month']==payroll['month'].max()]
last_month_payroll.head(5)

month department  employee_name           position  paid
34 2023-03-01      Sales       John Doe      Sales Manager  1500
35 2023-03-01      Sales     Jane Smith    Sales Associate   600
36 2023-03-01      Sales      Jim Brown    Sales Associate   700
37 2023-03-01      Sales   Laura Miller    Sales Associate   800
38 2023-03-01  Marketing  Alice Johnson  Marketing Manager  1650

In [ ]:
last_month_payroll_mkt = last_month_payroll[last_month_payroll['department'].isin(['Sales','Marketing'])]['paid'].sum()
last_month_payroll_mkt

np.int64(5950)

In [ ]:
# totalexpenses
totalexpenses += last_month_payroll_mkt
totalexpenses

np.int64(7650)

In [ ]:
daily_marketing_expendings = pd.read_excel('https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/export?format=xlsx',
                                   sheet_name='Sheet1')
daily_marketing_expendings.head(5)


date       channel  spending
0 2023-01-01    Google Ads       784
1 2023-01-01  Facebook Ads       659
2 2023-01-01  LinkedIn Ads       729
3 2023-01-01   Twitter Ads       292
4 2023-01-02    Google Ads       935

In [ ]:
last_marketing_expendings = daily_marketing_expendings[daily_marketing_expendings['date'].dt.month == daily_marketing_expendings['date'].max().month]['spending'].sum()
last_marketing_expendings

np.int64(68830)

In [ ]:
# last_marketing_expendings lọc theo tháng và ngày
#last_marketing_expendings = daily_marketing_expendings[(daily_marketing_expendings['date'].dt.month == daily_marketing_expendings['date'].max().month) & (daily_marketing_expendings['date'].dt.year == daily_marketing_expendings['date'].max().year)]

In [ ]:
totalexpenses += last_marketing_expendings
totalexpenses

np.int64(76480)

In [ ]:
receipt_history = pd.read_excel('https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/export?format=xlsx',
                             sheet_name='Sheet1')
receipt_history.head(5)

date  customer_id  receipt_amount  new_customer
0 2023-01-01         2653              67             1
1 2023-01-01         2731             271             1
2 2023-01-01         1277             231             0
3 2023-01-01         2094             107             0
4 2023-01-01         1314             416             0

In [ ]:
last_month_new_customer = receipt_history[(receipt_history['date'].dt.month == receipt_history['date'].max().month) & (receipt_history['new_customer']==1)]
last_month_new_customer.head(5)

date  customer_id  receipt_amount  new_customer
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
625 2023-03-01         2844             252             1
627 2023-03-01         2679             323             1
633 2023-03-02         1475              40             1

In [ ]:
# numbers_customers
number_customers = last_month_new_customer['customer_id'].nunique()
number_customers

63

In [ ]:
# cách 2:
#number_customers1 = len(last_month_new_customer)
#number_customers1

In [ ]:
# CAC
CAC = (totalexpenses / number_customers).item()
display()
print(f'CAC = {round(CAC,2)}')

CAC = 1213.97


## $\color{red}{\text{▶CAC = 1213.97}}$

## 2. Average Revenue Per User (ARPU)

$${APRU} = \frac{TotalRevenue}{NumberofUsers} $$


In [ ]:
# total_revenue
total_revenue = receipt_history[receipt_history['date'].dt.month == receipt_history['date'].max().month]['receipt_amount'].sum()
total_revenue

np.int64(83033)

In [ ]:
# number_of_users
number_of_users = receipt_history[receipt_history['date'].dt.month == receipt_history['date'].max().month]['customer_id'].nunique()
number_of_users

292

In [ ]:
# ARPU
ARPU = (total_revenue / number_of_users).item()
print(f'ARPU = {round(ARPU,2)}')

ARPU = 284.36


## $\color{red}{\text{▶ARPU = 284.36}}$

 ## 3. Cost of Goods Sold (COGS)

 $$ {COGS} = {BeginningInventory}+{PurchasesDuringthePeriod}-{EndingInventory} $$

#### Bởi vì đây là công ty về phần mềm nên 'Beginning Inventory' và 'EndingInventory' là 0.

In [ ]:
# last_month_item_expenses (Purchases During the Period)
production_item = ['AWS Hosting','Google Cloud Storage','Atlassian Jira']
last_month_item_expenses = last_month_expenses[last_month_expenses['item'].isin(production_item)]['amount'].sum()
last_month_item_expenses

np.int64(14200)

▶ 'Slack' và 'Zoom' ngoài được dùng cho team Software thì còn cho những team khác sử dụng tính biên độ cho sẽ là : $ 0.6 $

In [ ]:
# sharing_iteam (Purchases During the Period)
sharing_item = ['Slack','Zoom']
last_month_sharing_item = (last_month_expenses[last_month_expenses['item'].isin(sharing_item)]['amount'].sum())* 0.6
last_month_sharing_item

np.float64(864.0)

In [ ]:
# last_month_production_salary (Purchases During the Period)
last_month_production_salary = last_month_payroll[last_month_payroll['department'] == 'Engineering']['paid'].sum()
last_month_production_salary

np.int64(5200)

In [ ]:
# COGS
COGS = (last_month_item_expenses + last_month_sharing_item + last_month_production_salary).item()
print(f'COGS = {round(COGS,2)}')

COGS = 20264.0


## $\color{red}{\text{▶COGS = 20264.0}}$

## 4. Gross Margin

$${GrossMargin} = \frac{(Revenue - COGS)}{Revenue}*100 $$

In [ ]:
# GrossMargin
GrossMargin = ((total_revenue - COGS)/total_revenue) * 100
print(f'GrossMargin: {round(GrossMargin,2)}')

GrossMargin: 75.6


## $\color{red}{\text{▶GrossMargin = 75.6%}}$


## 5. Customer Lifetime Value (LTV)

$$ {LTV} = {ARPU}*{CustomerLifespan}*{GrossMargin} $$

In [ ]:
customer_lifespand = pd.read_excel('https://docs.google.com/spreadsheets/d/1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI/export?format=xlsx',
                         sheet_name='Sheet1')
customer_lifespand.head(5)

Unnamed: 0 start_date churn_date
0        1000 2021-11-15 2022-09-14
1        1001 2022-04-15 2023-02-16
2        1002 2022-10-30 2023-02-04
3        1003 2021-08-22 2023-02-07
4        1004 2021-08-23 2022-02-02

In [ ]:
customer_lifespand['lifespan_lifetime'] = (customer_lifespand['churn_date'] - customer_lifespand['start_date']).dt.days
customer_lifespand.head(5)

Unnamed: 0 start_date churn_date  lifespan_lifetime
0        1000 2021-11-15 2022-09-14                303
1        1001 2022-04-15 2023-02-16                307
2        1002 2022-10-30 2023-02-04                 97
3        1003 2021-08-22 2023-02-07                534
4        1004 2021-08-23 2022-02-02                163

In [ ]:
avg_lifespan_lifetime = customer_lifespand['lifespan_lifetime'].mean()
avg_lifespan_lifetime_month = avg_lifespan_lifetime / 30
avg_lifespan_lifetime_month

np.float64(9.841333333333333)

In [ ]:
# Bởi vì 'GrossMargin' có nhân với 100 -> phải chia lại cho 100
LTV = (ARPU * avg_lifespan_lifetime_month * GrossMargin) / 100
print(f'LTV = {round(LTV,2)}')

LTV = 2115.52


## $\color{red}{\text{▶LVT = 2115.52}} $

## 6. LTV/CAC



In [ ]:
LVT_CAC = LTV / CAC
print(f'LTV/CAC = {round(LVT_CAC,2)}')

LTV/CAC = 1.74


## $\color{red}{\text{▶LTV/CAC = 1.74}} $

## 7. Conclusion


### $\color{red}{\text {Nhận Định:}} $

**1. CAC:** \$ 1213.97 .

**2. ARPU:** \$ 284.36 .

**3. COGS:** \$ 20264.0 .

**4. GROSSMARGIN:** 75.6% $->$  Biên độ lợi nhuận tốt ($ >50\%\ $) .

**5. LVT:** \$ 2115.52 .

**6. LTV/CAC:** 1.74 $->$ Các chính sách Marketing và sản phẩm sản xuất ra đang đi đúng hướng , đúng đối tượng khách hàng.

#### **$->$** Tiếp tục thực hiện chiến lược kinh doanh của công ty, Marketing thêm nhằm tăng cường số lượng khách hàng mới, phát triển thêm phần mềm mới tới các khách hàng cũ -> nhằm tối ưu hơn về chính sách hậu mãi, cũng như tăng thêm lợi nhuận từng khách hàng lên.
